td-exception/src/interrupt.rs

```diff
@@ -424,7 +431,40 @@ interrupt_no_error!(virtualization, stack, {
             deadloop();
         }
     };
+
     stack.iret.rip += ve_info.exit_instruction_length as usize;
+
+    // If CET shadow stack is enabled, processor will compare the `LIP` value saved in the shadow stack and the
+    // `RIP` value saved in the normal stack when executing a return from an exception handler and cause a
+    // control protection exception if they do not match.
+    #[cfg(feature = "cet-shstk")]
+    {
+        // Read the Shadow Stack Pointer
+        let mut ssp: u64 = 0;
+        asm!(
+            "rdsspq {ssp}",
+            ssp = inout(reg) ssp,
+        );
+
+        // SSP -> Return address of current function | SSP | LIP | CS
+        let lip_ptr = ssp + 0x10;
+        let lip = stack.iret.rip;
+
+        // Enables the WRSSD/WRSSQ instructions by setting the `WR_SHSTK_E`
+        // to 1, then we can write the shadow stack
+        let mut msr_cet = Msr::new(MSR_IA32_S_CET);
+        msr_cet.write(msr_cet.read() | WR_SHSTK_E);
```

Comment:
We should not enable WR.

Please refer to https://github.com/tianocore/edk2/blob/master/MdePkg/Library/BaseLib/X64/LongJump.nasm#L49

Reply:
We need to change the `LIP` saved in `shadow stack memory`, not the shadow stack pointer. Change the shadow stack pointer doesn't affect the behavior returning from an exception (`iret`)